feflow/protocols/nonequilibrium_cycling.py

```diff
@@ -165,26 +166,32 @@ def _execute(self, ctx, *, state_a, state_b, mapping, settings, **inputs):
         # Note: by default this is cached to ctx.shared/db.json so shouldn't
         # incur too large a cost
         self.logger.info("Parameterizing molecules")
-        small_mols_a = []
-        for comp in state_a.components.values():
-            if isinstance(comp, SmallMoleculeComponent):
-                small_mols_a.append(comp)
-
-        for comp in small_mols_a:
-            offmol = comp.to_openff()
-            system_generator.create_system(offmol.to_topology().to_openmm(),
-                                           molecules=[offmol])
-            if comp == ligand_a:
-                mol_b = ligand_b.to_openff()
-                system_generator.create_system(mol_b.to_topology().to_openmm(),
-                                               molecules=[mol_b])
+        # TODO: Refactor if/when gufe provides the functionality https://github.com/OpenFreeEnergy/gufe/issues/251
+        # The following creates a dictionary with all the small molecules in the states, with the structure:
+        #    Dict[SmallMoleculeComponent, openff.toolkit.Molecule]
+        state_a_small_mols = {component: component.to_openff() for component in state_a.components.values() if
+                              isinstance(component, SmallMoleculeComponent)}
+        state_b_small_mols = {component: component.to_openff() for component in state_b.components.values() if
+                              isinstance(component, SmallMoleculeComponent)}
+
+        # Assign charges if unassigned -- more info: Openfe issue #576
+        for off_mol in chain(state_a_small_mols.values(), state_b_small_mols.values()):
+            # skip if we already have user charges
+            if not (off_mol.partial_charges is not None and np.any(off_mol.partial_charges)):
+                # due to issues with partial charge generation in ambertools
+                # we default to using the input conformer for charge generation
+                off_mol.assign_partial_charges(
+                    'am1bcc', use_conformers=off_mol.conformers
+                )
+            system_generator.create_system(off_mol.to_topology().to_openmm(),
+                                           molecules=[off_mol])
```

Comment:
Clarification question: are we just running `create_system` here to make sure that the charged molecules can be parameterized by OpenMM?

Reply:
This is for the solvation step a bit later on, if you don't do this here you don't have necessary topology info registered to solvate your system.